packages/wallets/backpack/src/adapter.ts

```diff
@@ -0,0 +1,259 @@
+import {
+    BaseMessageSignerWalletAdapter,
+    EventEmitter,
+    scopePollingDetectionStrategy,
+    SendTransactionOptions,
+    WalletAccountError,
+    WalletConnectionError,
+    WalletDisconnectedError,
+    WalletDisconnectionError,
+    WalletError,
+    WalletName,
+    WalletNotConnectedError,
+    WalletNotReadyError,
+    WalletPublicKeyError,
+    WalletReadyState,
+    WalletSignMessageError,
+    WalletSignTransactionError,
+    WalletWindowClosedError,
+} from '@solana/wallet-adapter-base';
+import { Signer, Connection, PublicKey, SendOptions, Transaction, TransactionSignature } from '@solana/web3.js';
+
+interface BackpackWalletEvents {
+    connect(...args: unknown[]): unknown;
+    disconnect(...args: unknown[]): unknown;
+}
+
+interface BackpackWallet extends EventEmitter<BackpackWalletEvents> {
+    isBackpack?: boolean;
+    publicKey?: { toBytes(): Uint8Array };
+    isConnected: boolean;
+    signTransaction(transaction: Transaction): Promise<Transaction>;
+    signAllTransactions(transactions: Transaction[]): Promise<Transaction[]>;
+    send(
+        transaction: Transaction,
+        signers?: Signer[],
+        options?: SendOptions,
+        connection?: Connection
+    ): Promise<TransactionSignature>;
+    signMessage(message: Uint8Array): Promise<Uint8Array | null>;
```

Comment:
What does it mean for this to return `null`?

Reply:
Moving this comment here https://github.com/solana-labs/wallet-adapter/pull/481#discussion_r928173263